gaphor/diagram/shapes.py

```diff
@@ -48,9 +57,88 @@ class Padding:  # Enum
     LEFT = 3
 
 
-def draw_border(box, context, bounding_box):
+DEFAULT_STYLE: Style = {
+    "min-width": 0,
+    "min-height": 0,
+    "padding": (0, 0, 0, 0),
+    "vertical-align": VerticalAlign.MIDDLE,
+    "vertical-spacing": 4,
+    "border-radius": 0,
+    "padding": (0, 0, 0, 0),
+    "background-color": None,
+    "line-width": 2,
+    "dash-style": [],
+    "color": None,
+    "font-family": "sans",
+    "font-size": 14,
+    "font-style": FontStyle.NORMAL,
+    "font-weight": None,
+    "text-decoration": None,
+    "text-align": TextAlign.CENTER,
+    "text-color": None,
+    "highlight-color": (0, 0, 1, 0.4),
+}
+
+
+@dataclass(frozen=True)
+class SizeContext:
+    cairo: CairoContext
+    style: Style
+
+
+@dataclass(frozen=True)
+class DrawContext:
+    """
+    Special context for draw()'ing the item. The draw-context contains
+    stuff like the cairo context and flags like selected and
+    focused.
+    """
+
+    cairo: CairoContext
+    selected: bool
+    focused: bool
+    hovered: bool
+    dropzone: bool
+    style: Style
+
+
+class cairo_state:
```

Comment:
I di not use `@contextmanager`, since this code is on the render path. It needs to be as lean as can be. I found lower case to be more appropriate since it behaves like a function, more than an object.

Reply:
Makes sense, thanks for the explanation!